similar_dev_search/model/repository_aggregator.py

```diff
@@ -0,0 +1,93 @@
+import asyncio
+from collections import Counter, defaultdict
+from typing import List, Tuple
+
+import httpx
+from tqdm import tqdm
+
+from model.constants import N_REPOS_FROM_STARGAZERS
+from model.developer import Developer
+from model.developer_entry import DeveloperEntry
+from model.repository import Repository
+
+
+class RepositoryAggregator:
+    def __init__(self, starting_repo: Repository):
+        self.starting_repo = starting_repo
+        self.top_repos = Counter()
+        self.developers_dict = None
+        self.developers_list = None
+        self.repos = None
+
+    async def get_developers(self, asyncio_client: httpx.AsyncClient = None) -> List[Developer]:
+        """
+        Returns a dict for each developer with their languages and variables
+        :param asyncio_client:
+        :return: dict of tuple of dicts - for languages and variables respectively
+        """
+        if self.developers_list is not None:
+            return self.developers_list
+
+        if asyncio_client is None:
+            client = httpx.AsyncClient(timeout=None)
+        else:
+            client = asyncio_client
+
+        self.developers_dict = defaultdict(DeveloperEntry)
+        tasks = []
+        for repo in tqdm((await self.get_repos(client))[:N_REPOS_FROM_STARGAZERS], "Analyzing repos"):
+            tasks.append(repo.get_developers())
+
+        result = await asyncio.gather(*tasks)
+
+        for gathered_devs in tqdm(result, "Processing developers: " + str(len(result))):
+            for developer, (languages, variables) in gathered_devs:
```

Comment:
for the future - each PR should be workable - right now it relies on future PRs so it will raise an error during runtime